.github/workflows/deploy.yml

```diff
@@ -0,0 +1,130 @@
+name: Deploy to Fargate
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment to deploy to'
+        required: true
+        type: choice
+        options:
+          - stage
+          - prod
+  repository_dispatch:
+    types: [deploy-stage, deploy-prod]
+  # Allow testing from feature branches
+  push:
+    branches:
+      - fargate
+
+permissions:
+  id-token: write
+  contents: read
+
+# Cancel in-progress deployments when a new one starts
+concurrency:
+  group: deploy-${{ github.event.inputs.environment || 'stage' }}
+  cancel-in-progress: true
+
+env:
+  AWS_REGION: us-west-2
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment: ${{ github.event.inputs.environment || (github.event.action == 'deploy-prod' && 'prod' || 'stage') }}
+
+    steps:
+      - name: Determine environment
+        id: env
+        run: |
+          if [ -n "${{ github.event.inputs.environment }}" ]; then
+            ENV="${{ github.event.inputs.environment }}"
+          elif [ "${{ github.event.action }}" = "deploy-prod" ]; then
+            ENV="prod"
+          else
+            ENV="stage"
+          fi
+          echo "environment=$ENV" >> $GITHUB_OUTPUT
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::768512802988:role/thunderbird-website-deploy
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Login to ECR
+        id: ecr-login
+        uses: aws-actions/amazon-ecr-login@v2
+
+      - name: Create placeholders for deployment build
+        run: |
+          mkdir -p dist/www.thunderbird.net dist/start.thunderbird.net dist/updates.thunderbird.net dist/tb.pro media
+          touch dist/www.thunderbird.net/.gitkeep dist/start.thunderbird.net/.gitkeep
+          touch dist/updates.thunderbird.net/.gitkeep dist/tb.pro/.gitkeep media/.gitkeep
+
+      - name: Build and push Docker image
+        id: build-image
+        env:
+          ECR_REGISTRY: ${{ steps.ecr-login.outputs.registry }}
+          IMAGE_TAG: ${{ github.sha }}
+          BUILD_ENV: ${{ steps.env.outputs.environment }}
+        run: |
+          IMAGE_URI="$ECR_REGISTRY/thunderbird-website-$BUILD_ENV:$IMAGE_TAG"
+          docker build \
+            -f docker/Dockerfile \
+            --build-arg BUILD_ENV=$BUILD_ENV \
+            -t $IMAGE_URI \
+            -t $ECR_REGISTRY/thunderbird-website-$BUILD_ENV:latest \
+            .
+          docker push $IMAGE_URI
+          docker push $ECR_REGISTRY/thunderbird-website-$BUILD_ENV:latest
+          echo "image_uri=$IMAGE_URI" >> $GITHUB_OUTPUT
```

Comment:
Oh nice, we can start using BUILD_ENV on the website and move some client-side hacks to build-time.